File: sdk/python/hitechcloud/resources/affiliate.py
"""Affiliate resource (6 endpoints)"""

from . import BaseResource


class AffiliateResource(BaseResource):
    """Affiliate: affiliate program management"""

    def get_details(self) -> dict:
        """GET /api/affiliate - Get affiliate account details"""
        return self._http.get("/api/affiliate")

    def get_balance(self) -> str:
        """GET /api/affiliate/balance - Get affiliate balance"""
        return self._http.get("/api/affiliate/balance")

    def get_payout(self) -> dict:
        """GET /api/affiliate/payout - Get payout details"""
        return self._http.get("/api/affiliate/payout")

    def update_payout(self, **kwargs) -> dict:
        """PUT /api/affiliate/payout - Update payout details"""
        return self._http.put("/api/affiliate/payout", kwargs)

    def list_referrals(self, page: int = None, per_page: int = None) -> dict:
        """GET /api/affiliate/referrals - List referrals"""
        params = {}
        if page is not None:
            params["page"] = page
        if per_page is not None:
            params["per_page"] = per_page
        return self._http.get("/api/affiliate/referrals", params)

    def list_transactions(self, page: int = None, per_page: int = None) -> dict:
        """GET /api/affiliate/transactions - List affiliate transactions"""
        params = {}
        if page is not None:
            params["page"] = page
        if per_page is not None:
            params["per_page"] = per_page
        return self._http.get("/api/affiliate/transactions", params)

File: sdk/ruby/lib/hitechcloud/resources/affiliate.rb
module HiTechCloud
  # Affiliate resource
  class AffiliateResource < BaseResource

    # Get affiliate summary
    def summary
      @http.get("/api/affiliates/summary")
    end

    # Get campaigns
    def campaigns
      @http.get("/api/affiliates/campaigns")
    end

    # Get commissions
    def commissions
      @http.get("/api/affiliates/commissions")
    end

    # Get payouts
    def payouts
      @http.get("/api/affiliates/payouts")
    end

    # Get vouchers
    def vouchers
      @http.get("/api/affiliates/vouchers")
    end

    # Get commission plans
    def commission_plans
      @http.get("/api/affiliates/commissionplans")
    end

  end
end
